File: structures-js/structures-api/src/api/idl/TenantSelectionC3Type.ts
import {ArrayC3Type, StringC3Type} from '@kinotic/continuum-idl'

/**
 * Represents a {@link TenantSelection} type.
 * This is used to denote when a tenant selection is desired by a named query.
 */
export class TenantSelectionC3Type extends ArrayC3Type {

    public readonly type: string = 'tenantSelection'

    constructor() {
        super(new StringC3Type())
    }
}

File: structures-api/src/main/java/org/kinotic/structures/api/domain/idl/TenantSelectionC3Type.java
package org.kinotic.structures.api.domain.idl;

import org.kinotic.continuum.idl.api.schema.ArrayC3Type;
import org.kinotic.continuum.idl.api.schema.StringC3Type;

/**
 * Represents the TenantSelection Typescript type
 * Created By Navíd Mitchell 🤪on 2/21/25
 */
public class TenantSelectionC3Type extends ArrayC3Type {

    public TenantSelectionC3Type() {
        super(new StringC3Type());
    }
}
